src/controllers/userController.ts

```diff
@@ -62,4 +63,75 @@ const login = async (
   res.status(StatusCodes.OK).json({ user: { name: user.name }, token });
 };
 
-export { register, login };
+const getUserProfile = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
```

Comment:
if user is authenticated then you have this in the user variable on line 72, 
you don't need to use any input from req.body, that is a vulnerability

Reply:
fixed